neo/io/nwbio.py

```diff
@@ -0,0 +1,811 @@
+"""
+NWBIO
+========
+
+IO class for reading data from a Neurodata Without Borders (NWB) dataset
+
+Documentation : https://neurodatawithoutborders.github.io
+Depends on: h5py, nwb, dateutil
+Supported: Read, Write
+Specification - https://github.com/NeurodataWithoutBorders/specification
+Python APIs - (1) https://github.com/AllenInstitute/nwb-api/tree/master/ainwb
+	          (2) https://github.com/AllenInstitute/AllenSDK/blob/master/allensdk/core/nwb_data_set.py
+              (3) https://github.com/NeurodataWithoutBorders/api-python
+Sample datasets from CRCNS - https://crcns.org/NWB
+Sample datasets from Allen Institute - http://alleninstitute.github.io/AllenSDK/cell_types.html#neurodata-without-borders
+"""
+
+from __future__ import absolute_import, division
+from neo.core import baseneo
+
+import os
+from itertools import chain
+from datetime import datetime
+import json
+try:
+    from json.decoder import JSONDecodeError
+except ImportError:  # Python 2
+    JSONDecodeError = ValueError
+from collections import defaultdict
+
+import numpy as np
+import quantities as pq
+from neo.io.baseio import BaseIO
+from neo.io.proxyobjects import (
+    AnalogSignalProxy as BaseAnalogSignalProxy,
+    EventProxy as BaseEventProxy,
+    EpochProxy as BaseEpochProxy,
+    SpikeTrainProxy as BaseSpikeTrainProxy
+)
```

Comment:
I plan to make a BaseAnalogSignalProxy very soon.
Maybe we should wait for this.

Reply:
@samuelgarcia  This PR has priority over the new BaseAnalogSignalProxy implementation, no?

Reply:
In fact, the BaseAnalogSignalproxy could be later moved from here to a central location...